src/ExadelMentorship.BusinessLogic/MainJob.cs

```diff
@@ -0,0 +1,52 @@
+﻿using ExadelMentorship.BusinessLogic.Exceptions;
+using ExadelMentorship.BusinessLogic.Features;
+using ExadelMentorship.BusinessLogic.Features.WeatherFeature;
+using ExadelMentorship.BusinessLogic.Models;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace ExadelMentorship.BusinessLogic
+{
+    public class MainJob
+    {
+        IRWOperation _rwOperation;
+        public MainJob(IRWOperation rwOperation)
+        {
+            _rwOperation = rwOperation;
+        }
+
+        public City GetCityFromInput()
+        {
+            var inputedLine = _rwOperation.ReadLine();
+            return new City
+            {
+                Name = inputedLine
+            };
+        }
+
+        public async Task Execute(Weather weather)
```

Comment:
1. Use constructor injection for `Weather` dependency.
2. Cover `Execute` with unit tests.